### PR TITLE
feat: require all table cells to be non-empty

### PR DESCRIPTION
**Pull request by opqdonut**
for #2551, follow-up to #2556

# Checklist for author

Remove items that aren't applicable, check items that are done.

## Reviewability
- [x] Link to issue
- [x] Consider adding screenshots for ease of review

## Testing
- [x] Complex logic is unit tested
- [x] Valuable features are integration / browser / acceptance tested automatically

### CODE REVIEW

**Comment by opqdonut** [comment]
table-validation

File: src/clj/rems/form_validation.clj [hunk 1]
```diff
@@ -4,16 +4,28 @@
             [rems.common.form :as form]
             [rems.common.util :refer [+email-regex+]]))
 
+(defn- all-columns-set? [field]
+  (let [valid-row? (fn [cells] (every? #(not (str/blank? (:value %))) cells))]
```
Comment (Macroz) [inline src/clj/rems/form_validation.clj:8]: `not-any?`? I think it reads slightly better here.

**Review by Macroz — APPROVED** [review]
I think the validation functions vs. predicates was a good call :+1:

File: src/clj/rems/form_validation.clj (hunk 1 above)
Comment (opqdonut) [reply src/clj/rems/form_validation.clj:8]: yeah!

**Pull request merged**